From 592c578cfa2842991c0e9611e04bcd01ebca950c Mon Sep 17 00:00:00 2001
From: Haihao Xiang <haihao.xiang@intel.com>
Date: Wed, 12 Feb 2020 18:08:22 +0800
Subject: [PATCH 1/3] Set the value for unknown or unsupported attributes to
 VA_ATTRIB_NOT_SUPPORTED

It is the requirement of VA-API

See https://github.com/intel/libva/blob/master/va/va.h#L1187-L1194
---
 media_driver/linux/gen10/ddi/media_libva_caps_g10.cpp | 1 +
 media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp | 1 +
 media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp | 1 +
 media_driver/linux/gen8/ddi/media_libva_caps_g8.cpp   | 1 +
 media_driver/linux/gen9/ddi/media_libva_caps_g9.cpp   | 1 +
 5 files changed, 5 insertions(+)

diff --git a/media_driver/linux/gen10/ddi/media_libva_caps_g10.cpp b/media_driver/linux/gen10/ddi/media_libva_caps_g10.cpp
index 8fbf810a..241b80e1 100755
--- a/media_driver/linux/gen10/ddi/media_libva_caps_g10.cpp
+++ b/media_driver/linux/gen10/ddi/media_libva_caps_g10.cpp
@@ -129,6 +129,7 @@ VAStatus MediaLibvaCapsG10::GetPlatformSpecificAttrib(VAProfile profile,
 {
     DDI_CHK_NULL(value, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     VAStatus status = VA_STATUS_SUCCESS;
+    *value = VA_ATTRIB_NOT_SUPPORTED;
     switch ((int)type)
     {
         case VAConfigAttribEncMaxRefFrames:
diff --git a/media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp b/media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp
index eb5c3d25..6c6671f6 100755
--- a/media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp
+++ b/media_driver/linux/gen11/ddi/media_libva_caps_g11.cpp
@@ -272,6 +272,7 @@ VAStatus MediaLibvaCapsG11::GetPlatformSpecificAttrib(VAProfile profile,
 {
     DDI_CHK_NULL(value, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER); 
     VAStatus status = VA_STATUS_SUCCESS;
+    *value = VA_ATTRIB_NOT_SUPPORTED;
     switch ((int)type)
     {
         case VAConfigAttribEncMaxRefFrames:
diff --git a/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp b/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
index 591a02a0..8c30b612 100644
--- a/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
+++ b/media_driver/linux/gen12/ddi/media_libva_caps_g12.cpp
@@ -318,6 +318,7 @@ VAStatus MediaLibvaCapsG12::GetPlatformSpecificAttrib(VAProfile profile,
 {
     DDI_CHK_NULL(value, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     VAStatus status = VA_STATUS_SUCCESS;
+    *value = VA_ATTRIB_NOT_SUPPORTED;
     switch ((int)type)
     {
         case VAConfigAttribEncMaxRefFrames:
diff --git a/media_driver/linux/gen8/ddi/media_libva_caps_g8.cpp b/media_driver/linux/gen8/ddi/media_libva_caps_g8.cpp
index 8784202d..e182d4af 100644
--- a/media_driver/linux/gen8/ddi/media_libva_caps_g8.cpp
+++ b/media_driver/linux/gen8/ddi/media_libva_caps_g8.cpp
@@ -126,6 +126,7 @@ VAStatus MediaLibvaCapsG8::GetPlatformSpecificAttrib(
 {
     DDI_CHK_NULL(value, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     VAStatus status = VA_STATUS_SUCCESS;
+    *value = VA_ATTRIB_NOT_SUPPORTED;
     switch ((int32_t)type)
     {
         case VAConfigAttribEncMaxRefFrames:
diff --git a/media_driver/linux/gen9/ddi/media_libva_caps_g9.cpp b/media_driver/linux/gen9/ddi/media_libva_caps_g9.cpp
index f49cfec0..1bb9f252 100755
--- a/media_driver/linux/gen9/ddi/media_libva_caps_g9.cpp
+++ b/media_driver/linux/gen9/ddi/media_libva_caps_g9.cpp
@@ -133,6 +133,7 @@ VAStatus MediaLibvaCapsG9::GetPlatformSpecificAttrib(VAProfile profile,
 {
     DDI_CHK_NULL(value, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     VAStatus status = VA_STATUS_SUCCESS;
+    *value = VA_ATTRIB_NOT_SUPPORTED;
     switch ((int)type)
     {
         case VAConfigAttribEncMaxRefFrames:
-- 
2.17.1

